### ISSUE TITLE
minSdkVersion for the ServiceStack Android client

### ISSUE THREAD

**Issue by bramley-turner-stride** [ServiceStack/ServiceStack.Java#19]
The ServiceStack Android client currently has its `minSdkVersion` set to `15`. I want to be able to use it back to sdk `11` (Honeycomb), so I wanted to know what (if any) features would not work. I opened the library, set the minSdkVersion to 11, and ran the `Calling new methods on older versions` lint check (Analyze->Run inspection by name). It didn't find anything, so it looks like there would be no issue lowering it to 11. It looks like the only 'newer' feature used is `new CookieManager()` in `JsonServiceClient`, which needs sdk 9 (Gingerbread).

Could you lower the minSdkVersion in the Android client library to the minimum version the library will actually work on?

Thanks

**Comment by mythz** [comment]
Starting a new project in Android Studio defaults to `minSdkVersion 16` and says it will run on approximately 95.2% of devices, I tried to change it to 11 but resulted in a build error, so I'm leaving it as is.

**Comment by bramley-turner-stride** [comment]
Hi,
Thanks for looking at this. Not sure I've explained it properly, let me try again:
I'm talking about the `minSdkVersion` defined in the AndroidClient library, here:
https://github.com/ServiceStack/ServiceStack.Java/blob/master/src/AndroidClient/android/build.gradle
This library also doesn't appear to need the dependency `com.android.support:appcompat-v7:23.1.1`, which could be safely deleted.

The entire project still builds fine with these changes.

(not talking about the `servicestack.net.androidclient` "app" demo application, https://github.com/ServiceStack/ServiceStack.Java/blob/master/src/AndroidClient/app/build.gradle which probably _does_ need minSdkVersion 16)

Why this matters:
When I add `compile 'net.servicestack:android:1.0.24'` to my app, a "manifest merger" occurs, where it checks the minSdkVersion of my app's build.gradle is >= the highest defined in any dependency. My app is compatible down to 11 - Honeycomb. When I add the servicestack library, I get a build error (minSdk incompatible). I can override this check in my app's manifest, with:
`<uses-sdk        tools:overrideLibrary="net.servicestack.android"/>`, which solves the initial problem. But then I wonder what features the service stack library depends on that aren't available in sdk 11 (so I'd know to explicitly handle that) - and it turns out there are no issues.

Re the `com.android.support:appcompat-v7:23.1.1` dependency: the SS client library doesn't use this, so it can be deleted. It causes this library to be included in my app, which may not necessarily need it (this library weighs 880KB plus its own dependencies, so it would be a large saving if it could be removed).
I can override this myself too as a workaround, like this:
````
compile ('net.servicestack:android:1.0.24'){
        exclude group: 'com.android.support'
    }
````

**Comment by mythz** [comment]
ok cook if it saves on merge conflicts we can remove it, I've also removed appcompat dependency as suggested in this commit: https://github.com/ServiceStack/ServiceStack.Java/commit/22b89c9546cb13251fc945a49a6f721220270ec5
